src/osd/PGStateUtils.cc

```diff
@@ -10,6 +10,7 @@ NamedState::NamedState(PGStateHistory *pgsh, const char *state_name)
   if(pgsh) {
     pgsh->enter(ceph_clock_now(), state_name);
   }
+  enter_time = ceph_clock_now();
```

Comment:
could you instead initialize `enter_time` with initialize list, and use it for updating `pgsh` to be consistent ? like
```c++
NamedState::NamedState(PGStateHistory *pgsh, const char *state_name)
  : pgsh(pgsh), state_name(state_name), enter_time(ceph_clock_now()) {
  if (pgsh) {
    pgsh->enter(enter_time, state_name);
  }
}
```

also might want to note, this is a regression introduce by d96c9072b72c7e8cf3757dc3f28afeebdce1cf66